test/framework/src/main/java/org/elasticsearch/search/geo/GeoBoundingBoxQueryIntegTestCase.java

```diff
@@ -31,19 +30,20 @@
 import static org.hamcrest.Matchers.anyOf;
 import static org.hamcrest.Matchers.equalTo;
 
-abstract class AbstractGeoBoundingBoxQueryIT extends ESIntegTestCase {
+public abstract class GeoBoundingBoxQueryIntegTestCase extends ESIntegTestCase {
+
+    public abstract XContentBuilder getMapping() throws IOException;
+
+    public abstract Version getVersion() throws IOException;
```

Comment:
I think this method can use some javadoc, it is not very clear what version we are talking about. It might also help to rename it into something like `randomSupportedVersion`.